src/model/parseModelDefinition.ts

```diff
@@ -58,6 +60,14 @@ export function parseModelDefinition<Dictionary extends ModelDictionary>(
         return result
       }
 
+      if (isStrictlyObject(definition[property])) {
+        const flattenedProperties = Object.keys(
+          flatten(definition[property]),
```

Comment:
Can we utilize [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) here instead?